src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -121,4 +128,31 @@ public static Email parseEmail(String email) throws ParseException {
         }
         return new Email(trimmedEmail);
     }
+
+    /**
+     * Parses a {@code String tag} into a {@code Tag}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code tag} is invalid.
+     */
+    public static FreeTimeTag parseFreeTimeTag(String tag) throws ParseException {
+        requireNonNull(tag);
+        String trimmedTag = tag.trim();
+        if (!FreeTimeTag.isValidTagName(trimmedTag)) {
+            throw new ParseException(FreeTimeTag.MESSAGE_CONSTRAINTS);
+        }
+        return new FreeTimeTag(trimmedTag);
+    }
+
+    /**
+     * Parses {@code Collection<String> tags} into a {@code Set<Tag>}.
+     */
+    public static Set<FreeTimeTag> parseFreeTimeTags(Collection<String> tags) throws ParseException {
+        requireNonNull(tags);
+        final Set<FreeTimeTag> tagSet = new HashSet<>();
+        for (String tagName : tags) {
+            tagSet.add(parseFreeTimeTag(tagName));
```

Comment:
Can we show the multiple tags for one day? Since a single day may have multiple free time periods.

Reply:
Yeap we can